src/hooks/simulation/useGetDifference.ts

```diff
@@ -0,0 +1,80 @@
+import { useCurrentSimulation, useFormState, useRule } from '@/publicodes-state'
+import type { Metrics } from '@incubateur-ademe/nosgestesclimat'
+import { useEffect, useRef, useState } from 'react'
+
+export function useGetDifference({ metric }: { metric: Metrics }): {
+  difference: number
+  updateKey: number
+} {
+  const { numericValue } = useRule('bilan', metric)
+
+  const [difference, setDifference] = useState(0)
+  const [countValueUpdate, setCountValueUpdate] = useState(0)
+  // We need this value to force the component to re-render when the numericValue changes
+  // We don't use numericValue directly because it update before the displayDifference
+  const [keyFromNumericValue, setKeyFromNumericValue] = useState(numericValue)
+  const [hasAlreadyAnswered, setHasAlreadyAnswered] = useState<
+    boolean | undefined
+  >(undefined)
+
+  const { foldedSteps } = useCurrentSimulation()
+
+  const { currentQuestion } = useFormState()
+
+  const prevValue = useRef(numericValue)
+  const prevQuestion = useRef(currentQuestion)
+
+  // Reset component state upon question change
+  useEffect(() => {
+    if (prevQuestion.current !== currentQuestion) {
+      setDifference(0)
+      setHasAlreadyAnswered(undefined)
+      setCountValueUpdate(0)
+    }
+  }, [currentQuestion])
+
+  // Dynamically set hasAlreadyAnswered when user visits question he/she has already answered
+  useEffect(() => {
+    const isInFoldedSteps = !!(
+      foldedSteps &&
+      currentQuestion &&
+      foldedSteps.includes(currentQuestion)
+    )
+
+    if (typeof hasAlreadyAnswered === 'undefined') {
```

Comment:
Why not update every time ?

Reply:
J'ajoute ce commentaire dans le code : 
```
    // This is set only once, at mount or when the question changes,
    // after getting the foldedSteps from the local state
```

Reply:
On veut que la valeur soit à true ou false uniquement si la valeur du useState est à undefined (mount) ou si on est passés à la question suivante (reset à undefined), sans quoi mon useEffect suivant se déclenche inopinément